src/io/EngineLogAccess.cpp

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#include "EngineLogAccess.hpp"
+#include "io/FileLineReader.hpp"
+#include "io/KeyValueFileReader.hpp"
+#include "io/KeyValueFileWriter.hpp"
+#include "LocalPath.hpp"
+
+unsigned
+EngineLogAccess::ReadLog()
+{
+  try{
+    FileLineReaderA file(LocalPath(log_file_name));
+    KeyValueFileReader kvreader(file);
+    KeyValuePair pair;
+    while (kvreader.Read(pair)) {
+      std::string bt_mac(pair.key);
+      double engine_runtime = std::atof(pair.value);
+      SensorToLog(bt_mac, engine_runtime);
+    }
+  } catch (...){}
+  return engine_log.size();
+}
+
+bool
+EngineLogAccess::WriteLog()
+{
+  try{
+    FileOutputStream file(LocalPath(log_file_name), Mode::CREATE);
+    BufferedOutputStream buffered(file);
+    KeyValueFileWriter kvwriter(buffered);
+
+    for (auto const& elem : engine_log) {    
+      kvwriter.Write(elem.first.c_str(), std::to_string(elem.second).c_str());    
+    }
+
+    buffered.Flush();
+    file.Commit();
+  } catch (...){
+    return false;
+  }
+  return true;
+}
+
+bool
+EngineLogAccess::WriteSingleUpdateToLog(std::string bt_mac, double engine_runtime)
+{
+  SensorToLog(bt_mac, engine_runtime);
+
+  try{
+    FileOutputStream file(LocalPath(log_file_name), Mode::APPEND_OR_CREATE);
+    BufferedOutputStream buffered(file);
+    KeyValueFileWriter kvwriter(buffered);
+    kvwriter.Write(bt_mac.c_str(), std::to_string(engine_runtime).c_str());    
+    buffered.Flush();
```

Comment:
This just creates big amounts of overhead for no reason. The buffer is not needed because there will be only one write, and the KeyValueFileWriter is useless here, and `std::to_string` has more useless overhead for allocating another buffer on the heap.
Why not just
```
const auto line = FmtBuffer<64>("{}={}\n", bt_mac, engine_runtime);
file.Write(line, strlen(line));
```
A very small fixed-size buffer allocation on the stack where libfmt produces the string, and a single `write()` system call from that buffer.